src/ocean/meta/traits/Basic.d

```diff
@@ -400,6 +400,43 @@ unittest
 
 *******************************************************************************/
 
+template isReferenceType ( T )
+{
+    const isReferenceType =
+           isPointerType!(T)
+        || isFunctionType!(T)
```

Comment:
Only function *pointers* are reference types, not functions, so this would yield a false positive if a function type is passed. (I can't think of a reasonable use case for this, but it can at least hide a bug in the code using this template.)

(Personal note: Function types are the most obscure and unnecessary D types I know and should IMHO be removed from the language because while being useless they need special handling like this.)